test/integration/compoundWrapModule.spec.ts

```diff
@@ -0,0 +1,200 @@
+    await setup.controller.addModule(wrapModule.address);
+
+    // compoundWrapAdapter setup
+    const compoundLibrary = await deployer.libraries.deployCompound();
+    const compoundWrapAdapter = await deployer.adapters.deployCompoundWrapAdapter("Compound", compoundLibrary.address);
+    await setup.integrationRegistry.addIntegration(wrapModule.address, compoundWrapAdapterIntegrationName, compoundWrapAdapter.address);
+  });
+
+  addSnapshotBeforeRestoreAfterEach();
+
+  context("when a SetToken has been deployed and issued", async () => {
+    let setToken: SetToken;
+    let setTokensIssued: BigNumber;
+
+    before(async () => {
+      setToken = await setup.createSetToken(
+        [setup.dai.address],
+        [ether(1)],
+        [setup.issuanceModule.address, wrapModule.address]
+      );
+
+      // Initialize modules
+      await setup.issuanceModule.initialize(setToken.address, ADDRESS_ZERO);
+      await wrapModule.initialize(setToken.address);
+
+      // Issue some Sets
+      setTokensIssued = ether(10);
+      const underlyingRequired = setTokensIssued;
+      await setup.dai.approve(setup.issuanceModule.address, underlyingRequired);
+      await setup.issuanceModule.issue(setToken.address, setTokensIssued, owner.address);
+    });
+
+    describe("#wrap", async () => {
+      let subjectSetToken: Address;
+      let subjectUnderlyingToken: Address;
+      let subjectWrappedToken: Address;
+      let subjectUnderlyingUnits: BigNumber;
+      let subjectIntegrationName: string;
+      let subjectCaller: Account;
+
+      beforeEach(async () => {
+        subjectSetToken = setToken.address;
+        subjectUnderlyingToken = setup.dai.address;
+        subjectWrappedToken = cDai.address;
+        subjectUnderlyingUnits = ether(1);
+        subjectIntegrationName = compoundWrapAdapterIntegrationName;
+        subjectCaller = owner;
+      });
+
+      async function subject(): Promise<any> {
+        return wrapModule.connect(subjectCaller.wallet).wrap(
+          subjectSetToken,
+          subjectUnderlyingToken,
+          subjectWrappedToken,
+          subjectUnderlyingUnits,
+          subjectIntegrationName,
+        );
+      }
+
+      it("should reduce the underlying quantity and mint the wrapped asset to the SetToken", async () => {
+        const previousUnderlyingBalance = await setup.dai.balanceOf(setToken.address);
+        const previousWrappedBalance = await cDai.balanceOf(setToken.address);
+
+        await subject();
+
+        const underlyingBalance = await setup.dai.balanceOf(setToken.address);
+
+        const wrappedBalance = await cDai.balanceOf(setToken.address);
+
+        const expectedUnderlyingBalance = previousUnderlyingBalance.sub(setTokensIssued);
+        expect(underlyingBalance).to.eq(expectedUnderlyingBalance);
+
+        const expectedWrappedBalance = previousWrappedBalance.add(setTokensIssued);
```

Comment:
Do these tests pass? Seems like there would be a mismatch of decimals since setTokens are `10**18` and cTokens are `10**8`

Reply:
It works because it is subtracting DAI and Set Token, both have 10**18